ts/src/sourcemapper/sourcemapper.ts

```diff
@@ -0,0 +1,249 @@
+   * of this mapper.
+   *
+   * @param {string} inputPath The path to an input file that could
+   *  possibly be the input to a transpilation process.  The path should be
+   *  relative to the process's current working directory.
+   */
+  hasMappingInfo(inputPath: string): boolean {
+    return this.getMappingInfo(inputPath) !== null;
+  }
+
+  /**
+   * @param {string} inputPath The path to an input file that could possibly
+   *  be the input to a transpilation process.  The path should be relative to
+   *  the process's current working directory
+   * @param {number} The line number in the input file where the line number is
+   *   zero-based.
+   * @param {number} (Optional) The column number in the line of the file
+   *   specified where the column number is zero-based.
+   * @return {Object} The object returned has a "file" attribute for the
+   *   path of the output file associated with the given input file (where the
+   *   path is relative to the process's current working directory),
+   *   a "line" attribute of the line number in the output file associated with
+   *   the given line number for the input file, and an optional "column" number
+   *   of the column number of the output file associated with the given file
+   *   and line information.
+   *
+   *   If the given input file does not have mapping information associated
+   *   with it then null is returned.
+   */
+  mappingInfo(inputPath: string, lineNumber: number, colNumber: number):
+      MapInfoOutput|null {
+    inputPath = path.normalize(inputPath);
+    const entry = this.getMappingInfo(inputPath);
+    if (entry === null) {
+      return null;
+    }
+
+    const sourcePos = {
+      source: path.relative(path.dirname(entry.mapFile), inputPath)
+                  .replace(/\\/g, '/'),
+      line: lineNumber + 1,  // the SourceMapConsumer expects the line number
+                             // to be one-based but expects the column number
+      column: colNumber      // to be zero-based
+    };
+
+    // TODO: Determine how to remove the explicit cast here.
+    const consumer: sourceMap.SourceMapConsumer =
+        entry.mapConsumer as {} as sourceMap.SourceMapConsumer;
+    const allPos = consumer.allGeneratedPositionsFor(sourcePos);
+    /*
+     * Based on testing, it appears that the following code is needed to
+     * properly get the correct mapping information.
+     *
+     * In particular, the generatedPositionFor() alone doesn't appear to
+     * give the correct mapping information.
+     */
+    const mappedPos: sourceMap.Position =
+        consumer.generatedPositionFor(sourcePos);
+
+    return {
+      file: entry.outputFile,
+      line: mappedPos.line - 1,  // convert the one-based line numbers returned
+                                 // by the SourceMapConsumer to the expected
+                                 // zero-based output.
+      // TODO: The `sourceMap.Position` type definition has a `column`
```

Comment:
TODO is still relevant?

Reply:
TODO is relevant.
Removed this, and updated code.